### PR TITLE
[wasm] Fix Uint8ClampedArray.From(byte[]) invalid cast

### PR DESCRIPTION
**Pull request by kjpou1**
Fixes issue https://github.com/mono/mono/issues/13787

- Add new BindingsTestSuite.cs program to the Tests that will test for these cases so regressions do not slip in.
- Add TypedArrayTypeCode enum

### CODE REVIEW

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kg** [comment]
Seems fine other than the comments I made (and the one question you can just resolve after answering). Enum change is not required but is preferable. Maybe we make that a backlog task

File: sdks/wasm/binding_support.js [hunk 1]
```diff
@@ -431,6 +431,9 @@ var BindingSupportLib = {
 				case 14:
 					newTypedArray = new Float64Array(end - begin);
 					break;
+				case 15:  // This is a special case because the typed array is also byte[]
```
[the remark this answers is not in the document]
Comment (kjpou1) [reply sdks/wasm/binding_support.js]: Yes, that was the original problem with the issue.  Uint8Array and Uint8ClampedArray are both represented with byte[].

File: sdks/wasm/Makefile [hunk 2]
```diff
@@ -172,7 +172,8 @@ build-test-suite: bin/test-suite/mono.js
 $(BROWSER_TEST)/.stamp-browser-test-suite: $(DRIVER_CONF)/.stamp-build packager.exe $(WASM_FRAMEWORK)/.stamp-framework $(BROWSER_TEST_SOURCES)
 	$(CSC) $(CSC_FLAGS) /target:library -out:$(BROWSER_TEST)/HttpTestSuite.dll $(BCL_DEPS) $(WASM_FRAMEWORK_DEPS) $(BROWSER_TEST)/HttpTestSuite.cs 
 	$(CSC) $(CSC_FLAGS) /target:library -out:$(BROWSER_TEST)/WebSocketTestSuite.dll $(BCL_DEPS) $(WASM_FRAMEWORK_DEPS) $(BROWSER_TEST)/WebSocketTestSuite.cs 
-	mono --debug packager.exe -out=$(BROWSER_TEST)/publish --copy=IfNewer --template=$(BROWSER_TEST)/runtime.js $(BROWSER_TEST_ASSETS) $(BROWSER_TEST)/HttpTestSuite.dll $(BROWSER_TEST)/WebSocketTestSuite.dll
+	$(CSC) $(CSC_FLAGS) /target:library -out:$(BROWSER_TEST)/BindingsTestSuite.dll $(BCL_DEPS) $(WASM_FRAMEWORK_DEPS) /r:$(WASM_BCL_DIR)/Facades/System.Memory.dll $(BROWSER_TEST)/BindingsTestSuite.cs
```
[the remark this answers is not in the document]
Comment (kjpou1) [reply sdks/wasm/Makefile]: Yes. Forgot to mention that.

File: sdks/wasm/framework/src/WebAssembly.Bindings/Core/TypedArray.cs [hunk 3]
```diff
@@ -107,6 +107,8 @@ public unsafe static T From (ReadOnlySpan<U> span)
 			ValidateFromSource (span);
 
 			int type = (int)Type.GetTypeCode (typeof (U));
+			if (type == 6 && typeof(T) == typeof(Uint8ClampedArray))
```
[the remark this answers is not in the document]
Comment (kjpou1) [reply sdks/wasm/framework/src/WebAssembly.Bindings/Core/TypedArray.cs]: Also, added tests

**Comment by kjpou1** [comment]
@monojenkins build

**Comment by kjpou1** [comment]
@monojenkins build

**Comment by lewing** [comment]
@monojenkins build failed

**Pull request merged**